java/eye-icon.js

```diff
@@ -0,0 +1,31 @@
+ import pw from "./pwEvent.js";
+ 
+ const eyeIcon = document.querySelector('.eye-one');
+ const eyeIcon2 = document.querySelector('.eye-two');
+
+ // 눈 모양 아이콘 클릭 시 비밀번호 텍스트, 패스워드 형식 변경 및 아이콘 변경
+
+//const eyeClick = (event) => changeOnOff(event);
+
+/*const changeOnOff = (event) => {//
+  if (event.target.classList.contains('eye-off')) {
+    event.target.classList.remove('eye-off');
+    pw.passwordInput.type = 'password';
+  } else {
+    event.target.classList.add('eye-off');
+    pw.passwordInput.type = 'text';
+  }
+}*/
+function changeOnOff(inputElement, eyeIconElement) {
+  if (inputElement.type === 'password') {
+    inputElement.type = 'text';
+    eyeIconElement.classList.add('off');
+  } else {
+    inputElement.type = 'password';
+    eyeIconElement.classList.remove('off');
+  }
+}
```

Comment:
c: 두 요소를 첫번째, 두번째 인자로 받기보다는 토글(on/off 변경)할 요소 하나만을 받는 함수를 받는 것이 여러 요소가 있을 때보다 유연하게 대처할 수 있어요.

그러기 위해서는 눈 모양 아이콘 요소가 input 요소의 마스킹 상태를 알 수 있어야하는데, 해당 부분은 [css 선택자](https://developer.mozilla.org/ko/docs/Web/CSS/CSS_selectors)로 가져올 수 있습니다.

```css
/* type이 password인 password-input id 요소 바로 다음에 오는 eye 클래스네임을 가진 형제 요소를 선택 */
#password-input[type='password'] + .eye {
  background-image: url(/image/eye-off.png);
}
/* type이 text인 password-input id 요소 바로 다음에 오는 eye 클래스네임을 가진 형제 요소를 선택 */
#password-input[type='text'] + .eye {
  background-image: url(/image/eye-on.png);
}
```
```suggestion
function togglePasswordInputMasking(element) {
  if (element.type === 'password') {
    element.type = 'text';
  } else {
    element.type = 'password';
  }
}
```